SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -404,41 +412,51 @@ bool FLocalDeploymentManager::TryStopLocalDeploymentGracefully()
 		return TryStopLocalDeployment();
 	}
 
-	if (!StartLocalDeploymentShutDown())
+	StoppingDeployment.Lock();
+	try
```

Comment:
I am not very experience when it comes to using try catch clauses so I might be talking a load of rubbish. My understanding here is that we are catching a fatal exception in the program, ensuring we don't cause the crash referred to in UNR-5524 and then throwing the original fatal exception. This seems reasonable to me in the case that the fatal issue referred to in UNR-5524 has affects outside the running of the program. Otherwise the program is just going to fail anyway. 

Perhaps a better way of phrasing it is asking: Is the crash referred to in UNR-5524 more dangerous than the exceptions that can come from this logic? If so then this try catch seems useful. 

Let me know if I am missing something.

Reply:
We don't actually catch the crash, we re-throw it, the main reason for this is to make sure we unlock the critical section if we throw anywhere in the try section.

Reply:
Similar to a mutex lock guard but for a critical section: https://en.cppreference.com/w/cpp/thread/lock_guard

Reply:
Sure, I understand that but my point is what is the purpose of unlocking the critical section to then throw an exception that will not be handled?

Reply:
we don't leave the critical section locked.

Reply:
also the exception could be handled at any point, we don't know it is actually unhandled

Reply:
for all we know the exception is part of some perfectly valid non-crashing code.

Reply:
its a hygiene thing. If you are really interested in this subject I could talk you through it, its not the most intuitive mechanic.

Reply:
We should not use try/catch blocks.

Reply:
I know there is a culture of not using try / catch / exceptions in game dev but I do wonder why you say we shouldn't use them? Its not part of any style guide we use I am aware of. Worth formalising somewhere?

Reply:
I am not sure but I don't think we handle exceptions anywhere else in our code? Is there some other mechanism by which exceptions may be handled? I don't think we have plans to implement exception handling throughout our code. If an exception is non-crashing but it is not handled, will the program still be able to run? I understand that the code is "more clean" but I can't see what actual benefits we will see if the exception is unhandled. Perhaps is preferable in some way for a program to stop due to an exception rather than full out crash?

Reply:
I am not sure how I explain it better in a comment thread but basically the catch part is more about using critical section than any crash happening at this particular site. You want guarantees that you do not leave a critical section locked when you exit the scope of this function and you need to cover the exception case. This complexity is incidentally why a lot of people hate exceptions. As mentioned I could talk you through this in a call if you are interested and still confused.

Its quite academic at this point as we don't have to use try / catch here.